src/compiler/utilities.ts

```diff
@@ -1281,402 +1281,400 @@ export function getInternalEmitFlags(node: Node): InternalEmitFlags {
 export type ScriptTargetFeatures = ReadonlyMap<string, ReadonlyMap<string, string[]>>;
 
 /** @internal */
-export function getScriptTargetFeatures(): ScriptTargetFeatures {

```

Comment:
Instead of turning this into a global that's always calculated, you probably should instead write this as:

```ts
export const getScriptTargetFeatures = memoize(() => new Map({ ... }))
```

Then it will still be lazy, but only calculated once.

Reply:
Actually, you'll also want to write:

```ts
export const getScriptTargetFeatures = /* @__PURE__ */ memoize(() => new Map({ ... }))
```

See similar calls which I made lazy e.g. `createPrinterWithDefaults`. This is mainly for tree shaking reasons as the function declaration doesn't have side effects but the memoize call does.

Reply:
good points, I will change it

Reply:
done